src/io_layer_distribution.c

```diff
@@ -86,69 +86,62 @@ int io_normalize_profile(profile_t *profile) {
 
   if (start_sign == end_sign || start_sign == 0 || end_sign == 0) {
     // Profile has invalid shape (not exactly one zero in interval interior)
-    printf("ZSF: Invalid profile shape!\n");
+    printf("ZSF: Invalid profile shape! (Start and end should differ in sign)\n");
     return -1;
   }
   // Find z_zero.
-  while (index_below_zero < max_index &&
-         fsign(profile->relative_discharge_from_lock[index_below_zero + 1]) == start_sign) {
-    ++index_below_zero;
+  while (index_before_zero < max_index &&
+         fsign(profile->relative_discharge_from_lock[index_before_zero + 1]) == start_sign) {
+    ++index_before_zero;
   }
-  while (index_above_zero > 0 &&
-         fsign(profile->relative_discharge_from_lock[index_above_zero - 1]) == end_sign) {
-    --index_above_zero;
+  while (index_after_zero > 0 &&
+         fsign(profile->relative_discharge_from_lock[index_after_zero - 1]) == end_sign) {
+    --index_after_zero;
   }
 
-  if (index_above_zero != index_below_zero + 1) {
+  if (index_after_zero != index_before_zero + 1) {
     // Error if there's not exactly one index between positive and negative.
-    if (index_above_zero - index_below_zero - 1 != 1) {
-      printf("ZSF: Too many zeroes in profile? (index_above_zero=%d, index_below_zero=%d.)\n", index_above_zero,
-             index_below_zero);
-      return -1;
-    }
-    // Error if that value is not zero.
-    if (fabs(profile->relative_discharge_from_lock[index_below_zero + 1]) > DBL_EPSILON) {
-      printf("ZSF: Zero in profile not zero?\n");
+    if (index_after_zero - index_before_zero > 2) {
+      printf("ZSF: Invalid profile shape! Too many zero entries in profile?\n     "
+             "(index_before_zero=%d "
```

Comment:
I like the improved message, but before = after in this line.

Reply:
Fixed.